hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java

```diff
@@ -618,6 +607,59 @@ public boolean getReplicationManagerStatus() {
     return scm.getReplicationManager().isRunning();
   }
 
+  /**
+   * Get Datanode disk metrics (such as capacity, used) by ip or uuid.
+   *
+   * @param ipaddress
+   * @param uuid
+   * @return DatanodeDiskMetrics
+   * @throws IOException
+   */
+  @Override
+  public HddsProtos.DatanodeDiskMetrics getDatanodeDiskMetrics(String ipaddress,
+                                                         String uuid)
+      throws IOException {
+
+    // check admin authorisation
+    String remoteUser = getRpcRemoteUsername();
+    try {
+      getScm().checkAdminAccess(remoteUser);
+    } catch (IOException e) {
+      LOG.error("Authorisation failed", e);
+      throw e;
+    }
+
+    // get datanode by ip or uuid
+    DatanodeDetails node = null;
+    if (!Strings.isNullOrEmpty(uuid)) {
+      node = scm.getScmNodeManager().getNodeByUuid(uuid);
+    } else if (!Strings.isNullOrEmpty(ipaddress)) {
+      List<DatanodeDetails> nodes = scm.getScmNodeManager()
+          .getNodesByAddress(ipaddress);
```

Comment:
I think we should also handle the case where both are specified and both correspond to different datanodes.

Reply:
I guess we can send the usage info for both in this case.

Reply:
Letting ipaddress and UUID stay exclusive as per our discussion.